lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -1004,6 +1004,27 @@ impl ShardReplicaSet {
         shard.trigger_optimizers();
         true
     }
+
+    /// Returns the estimated size of all locally stored vectors in bytes.
+    pub(crate) async fn estimated_local_vector_storage_size(&self) -> usize {
+        self.local
+            .read()
+            .await
+            .as_ref()
+            .map(|i| match i {
+                Shard::Local(local) => local
+                    .segments()
+                    .read()
+                    .iter()
+                    .map(|i| i.1.get().read().info().vectors_size_bytes)
```

Comment:
Are we going to call `info()` on all segments for each request?
Have you checked how performant it is or am I needlessly worrying :)

Reply:
Benchmark results:

With 2 payload indices:
```
segment-info            time:   [285.36 ns 285.57 ns 285.96 ns]
```

With 3 payload indices:
```
segment-info            time:   [364.87 ns 365.16 ns 365.58 ns]
```

Without payload index:
```
segment-info            time:   [100.33 ns 100.49 ns 100.75 ns]
```

Without payload index and calling `.clone()`:
```
segment-info            time:   [93.023 ns 93.194 ns 93.426 ns]
```

The performance of `info()` depends on the amount of indexed fields, since the schema [gets cloned](https://github.com/qdrant/qdrant/blob/dev/lib/segment/src/segment/entry.rs#L436).
I think this is still fast enough for our use case.

Reply:
This definitely is expensive, partially because of all the locking that needs to happen and because it collects much more than we actually need.

I'm not confident this is worth it on each request, especially if a lot of other operations are going on at the same time. I think we should be looking at a different approach.

Maybe this works: a variable on the shard level holding the current size estimate, which we update on each flush.

Reply:
Original idea was to keep cache and update it every ~32 update operations. So info should not be called on each request

Reply:
Thanks for chiming in, I would prefer if we introduce the amortized approach right away :pray: